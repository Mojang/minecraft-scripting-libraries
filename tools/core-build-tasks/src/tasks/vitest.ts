import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
