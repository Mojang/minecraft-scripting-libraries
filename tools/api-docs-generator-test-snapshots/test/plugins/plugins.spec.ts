// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import path from 'path';
import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Plugins', () => {
    it('Dynamically loads markup generator from a plugin module', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['test-1'],
            configPath: path.resolve(__dirname, 'api-docs-generator.config.mjs'),
        });
    });
});
