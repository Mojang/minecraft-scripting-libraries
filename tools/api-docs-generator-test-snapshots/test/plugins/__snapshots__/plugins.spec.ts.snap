// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Plugins > Dynamically loads markup generator from a plugin module > # Of Files Generated 1`] = `1`;

exports[`Plugins > Dynamically loads markup generator from a plugin module > test-1/test-1.txt 1`] = `
"Output Message: This string is defined in the config file
"
`;
